### DISCUSSION TITLE
🚀 Feature Request: Pip & Poetry support for Python Workers

### DISCUSSION THREAD

**Discussion by deeprobin** [cloudflare/workerd#1988] [Python Packages]
### Describe the solution

As I have already learned on the Discord, there are plans to support the requirements.txt (pip packages).

What I personally am also missing is support for pyproject.toml (poetry) files.
I think that would also give you a small market advantage over Azure Functions & Co. Only pip packages are supported there.

Poetry is a more modern package manager than pip and is now being used more and more widely.
https://github.com/python-poetry/poetry

Since I did not find an issue for pip/requirements.txt, I would have suggested that this issue serves as a tracking issue for the package manager support.

cc @dom96

**Comment by euan-reid** [comment]
I'd like to note that `pyproject.toml` is not unique to poetry - it's a Python standard, and as of [PEP-621](https://peps.python.org/pep-0621/) it has a standard way to define dependencies. Perhaps this should be about supporting that instead? (#python-poetry/poetry/issues/3332 exists to migrate Poetry from their tool-specific `tool.poetry.dependencies` to the standard `dependencies` key)

**Comment by garysassano** [comment]
What about [uv](https://github.com/astral-sh/uv)? It's exploding in popularity, along with [ruff](https://github.com/astral-sh/ruff) (also from Astral).
